x-pack/plugins/security/server/lib/authentication/providers/__tests__/token.js

```diff
@@ -6,6 +6,7 @@
 
 import expect from 'expect.js';
```

Comment:
If I'm following along correctly, the following tests should be updated as well:
- `fails if state contains invalid credentials.` line 179
- `fails when session contains a rejected token` line 285

When submitting a request to `_security/_authenticate` with an invalid Bearer header, we're getting back a 401 as well, which will match the new `isAccessTokenExpiredError` and attempt to use the refresh token to get a new access token. Previously, we weren't trying to use the refresh tokens in this situation.

Reply:
> fails if state contains invalid credentials. line 179
> fails when session contains a rejected token line 285

Hmm, right, but If we change these tests, then they will be exactly the same as `succeeds with valid session even if requiring a token refresh`, line 134 or `redirects non-AJAX requests to /login and clears session if token refresh fails with 400 error'` line 360, depending on what happens during refreshing.

It feels like we can either remove these tests completely or change them to test non-401 errors (e.g. 500), what do you think?

Reply:
I like the idea of adding tests for the 500 situation

Reply:
Okay, will do then!